pkgs/build-support/flatten-references-graph/src/flatten_references_graph/split_paths.py

```diff
@@ -0,0 +1,226 @@
+from toolz import curried as tlz
+from toolz import curry
+
+from .lib import (
+    debug,
+    debug_plot,
+    DEBUG_PLOT,
+    find_vertex_by_name_or_none,
+    graph_is_empty,
+    is_None,
+    subcomponent_multi,
+    unnest_iterable
+)
+
+
+@curry
+def add_root(added_root_name, graph):
+    """Add single root to the graph connected to all existing roots.
+
+    If graph has only one root, return the graph unchanged and the name
+    of the root vertex.
```

Comment:
Weird API, right? Why does it behave differently when there's only one root? Maybe this function should always add the root, and the callee can decide when to call it.

Reply:
I think I should have named it: `coerce_to_singly_rooted_graph`, and then the implementation would make sense

Reply:
addressed in 29836710e3c22db0bdb3a9c672daad5e4358c950